pilot/pkg/networking/core/v1alpha3/name_table.go

```diff
@@ -119,6 +119,10 @@ func (configgen *ConfigGeneratorImpl) BuildNameTable(node *model.Proxy, push *mo
 			nameInfo.Namespace = svc.Attributes.Namespace
 			nameInfo.Shortname = svc.Attributes.Name
 		}
+
+		if ni, ok := out.Table[string(svc.Hostname)]; ok {
```

Comment:
I am not sure this is safe. We don't have a way to tell if there are two services for the same hostname because they are in one ServiceEntry, or if they are two service entries in different namespaces, etc.

Reply:
Thanks for you quick repose!
Is it correct to add two service entries with the same hosts? I have just created two service entries on 1.10(in different namespace), and istio do not give warns, but I found the hosts is resolved to the address of the service entry which is in one namespace(perhaps the namespace whose has an alphabetical order).
I have also created two service entries with same hosts in the same namespace, the behavior is similar.

Reply:
The intent is that `push.Services(node)` will return the correct service based on the Sidecar scoping. If there is no scope selecting only a single one then one is chosen (oldest)

Reply:
I think this is better than before, ServiceEntry has many known issues. Users can influence each other unintendedly by mistake. For example, set different addresses for the same hostname in different ses.
Just one point should deduplicate ips.